src/mondemand_statdb.erl

```diff
@@ -558,11 +619,25 @@ construct_stats_msg (AllKeys = [#mdkey {prog_id = ProgId, context = Context}|_],
                           end,
                           CollectTime).
 
+-define(TIME_UNIT_NATIVE_TO_SECONDS, ct_expand:term(erlang:convert_time_unit(1, seconds, native))).
+
+finalize_metric (InternalKey = #mdkey {type = gcounter}, Table) ->
+  case ets:lookup (Table, InternalKey) of
+    [] -> ok;
+    [#md_gcounter {value = CV, previous_value = PV, previous_time = PT}] ->
+      CT = erlang:monotonic_time(),
+      Rate = round((CV - PV) * ?TIME_UNIT_NATIVE_TO_SECONDS / max(1, (CT - PT))),
+      ets:update_element(Table, InternalKey,
+                         [ {#md_gcounter.previous_value, CV},
+                           {#md_gcounter.previous_time, CT},
+                           {#md_gcounter.rate, Rate} ])
+  end.
```

Comment:
Doesn't this need a clause like?
```erlang
finalize_metric(_,_) -> ok.
```
Otherwise won't it crash for anything not a gcounter?